src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java

```diff
@@ -306,9 +310,42 @@ private <T extends SchemaObject> Tags getCustomTags(
     JsonApiMetricsConfig.SortType sortType = getVectorTypeTag(commandContext, command);
     Tag sortTypeTag = Tag.of(jsonApiMetricsConfig.sortType(), sortType.name());
 
+    // --- Command Feature Usage Tags ---
+    if (command instanceof FeatureSource fs) {
+      commandContext.addFeatureUsage(fs.getFeatureUsage());
+    }
+    Tags commandFeatureTags = getCommandFeatureTags(commandContext.featureUsage());
+
     // --- Combine All Tags ---
     return Tags.of(
-        commandTag, tenantTag, errorTag, errorClassTag, errorCodeTag, vectorEnabled, sortTypeTag);
+            commandTag,
+            tenantTag,
+            errorTag,
+            errorClassTag,
+            errorCodeTag,
+            vectorEnabled,
+            sortTypeTag)
+        .and(commandFeatureTags);
+  }
+
+  /**
+   * Adds tags for all defined features. If a feature is present in FeatureUsage, its tag value is
+   * "true", otherwise "false".
+   */
+  private Tags getCommandFeatureTags(FeatureUsage featureUsage) {
+    Tags tags = Tags.empty();
+    Set<Feature> usedFeatures = Collections.emptySet();
+
+    if (featureUsage != null && featureUsage.getFeatures() != null) {
```

Comment:
Should this be ever called with `featureUsage == null`? If not, remove check.

But overall I think all of this logic of building `Tags` should be moved in `FeatureUsage` (whatever it will be named) -- as per my comments for that class.